osdk/src/config/manifest.rs

```diff
@@ -32,6 +32,9 @@ pub enum ProjectType {
 /// The osdk manifest from configuration file `OSDK.toml`.
 #[derive(Debug, Clone)]
 pub struct TomlManifest {
+    #[cfg(test)]
+    pub project_type: Option<ProjectType>,
+    #[cfg(not(test))]
     #[expect(dead_code)]
```

Comment:
Possible/Better to do something like `#[cfg_attr(not(CONFIG_X), expect(dead_code))]`? 

(https://docs.kernel.org/next/rust/coding-guidelines.html#lints)